src/V3Assert.cpp

```diff
@@ -286,13 +286,16 @@ class AssertVisitor final : public VNVisitor {
                  itemp = VN_AS(itemp->nextp(), CaseItem)) {
                 if (itemp->isDefault()) has_default = true;
             }
+            const string valFmt = "'" + cvtToStr(nodep->exprp()->dtypep()->widthMin()) + "'h%X'";
```

Comment:
This will break for non-numeric types.  %p would work, but %p are converted earlier.  Maybe do this only for numeric types... and test a unique case with strings?

Reply:
I think I missed this problem with #4892 so check that too, thanks.

Reply:
You are right. Thanks for letting me know.